sample-apps/nextjs-standalone/src/app/files/route.js

```diff
@@ -0,0 +1,68 @@
+import { NextResponse } from "next/server";
+import { exec } from "child_process";
+import { promisify } from "util";
+
+const execAsync = promisify(exec);
+
+export async function GET(request) {
+  let directory = new URL(request.url).searchParams.get("path") ?? ".";
+  if (!directory.length) {
+    directory = ".";
+  }
+
+  // This is a command injection vulnerability. Do not use this in production.
+  const result = await execAsync(`ls '${directory}'`);
+
+  let stdout, stderr;
+  if (typeof result === "string") {
+    stdout = result;
+    stderr = "";
+  } else {
+    stdout = result.stdout;
+    stderr = result.stderr;
+  }
+
+  if (stderr && stderr.length > 0) {
+    return NextResponse.error(new Error(stderr));
+  }
+
+  if (!stdout || stdout.length === 0) {
+    return NextResponse.json({ files: [] });
+  }
+
+  return NextResponse.json({
+    files: stdout.split("\n").filter((line) => line),
+  });
+}
```

Comment:
maybe good to add a route handler with a database query?

Reply:
In addition to or instead of Shell Injection?